x/ecocredit/client/marketplace/proposal.go

```diff
@@ -0,0 +1,86 @@
+package marketplace
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govclient "github.com/cosmos/cosmos-sdk/x/gov/client"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/rest"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+var AllowDenomProposalHandler = govclient.NewProposalHandler(TxAllowDenomProposal, func(context client.Context) rest.ProposalRESTHandler {
+	return rest.ProposalRESTHandler{
+		SubRoute: "",
+		Handler:  nil,
+	}
+})
+
+func TxAllowDenomProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "allow-denom-proposal [path_to_file.json] [flags]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a proposal to add a denom to the list of allowed denoms",
+		Long: strings.TrimSpace(`Submit a proposal to add a denom to the list of allowed denoms for use in the marketplace. 
+The json file MUST take the following form:
+{
+	"title": "some title",
+	"description": "some description",
+	"allowed_denom": {
+					"bank_denom": "uregen",
+					"display_denom": "regen",
+					"exponent": 18
+	}
+}
```

Comment:
A little formatting, `denom` not `allowed_denom`, and `exponent` should reflect the `u` SI prefix in the example.

```suggestion
{
    "title": "some title",
    "description": "some description",
    "denom": {
	"bank_denom": "uregen",
        "display_denom": "regen",
        "exponent": 6
    }
}
```